### PR TITLE
Refs #34730 - Delete foreman_container_port_t before we stop assigning it

### PR DESCRIPTION
**Pull request by upadhyeammit**

### CODE REVIEW

File: foreman-selinux-enable [hunk 1]
```diff
@@ -22,6 +22,16 @@ fi
 for selinuxvariant in targeted
 do
   if /usr/sbin/semodule -s $selinuxvariant -l >/dev/null; then
+    # Create port list cache
+    /usr/sbin/semanage port -E > $TMP_PORTS
+
+    # Remove previously defined container port
+    # We need to remove the port first before we stop assigning it
+    grep -E 'container_port_t' $TMP_PORTS | sed s/-a/-d/g >> $TMP_EXEC_BEFORE
```
Comment (evgeni) [inline foreman-selinux-enable]:
```suggestion
    grep -E 'foreman_container_port_t' $TMP_PORTS | sed s/-a/-d/g >> $TMP_EXEC_BEFORE
```

**Comment by evgeni** [comment]
So I tested this on a 3.2 EL8 box:

```console
[root@centos8-stream-foreman-3-2 ~]# dnf install http://koji.katello.org/kojifiles/packages/foreman-selinux/3.3.0/0.1.develop.20220419072414git2577699.el8/noarch/foreman-selinux-3.3.0-0.1.develop.20220419072414git2577699.el8.noarch.rpm
Last metadata expiration check: 0:09:48 ago on Fri 22 Apr 2022 11:31:06 AM UTC.
foreman-selinux-3.3.0-0.1.develop.20220419072414git2577699.el8.noarch.rpm                                                                                                                              133 kB/s |  53 kB     00:00    
Dependencies resolved.
=======================================================================================================================================================================================================================================
 Package                                            Architecture                              Version                                                                            Repository                                       Size
=======================================================================================================================================================================================================================================
Upgrading:
 foreman-selinux                                    noarch                                    3.3.0-0.1.develop.20220419072414git2577699.el8                                     @commandline                                     53 k

Transaction Summary
=======================================================================================================================================================================================================================================
Upgrade  1 Package

Total size: 53 k
Is this ok [y/N]: y
Downloading Packages:
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                                                                               1/1 
  Upgrading        : foreman-selinux-3.3.0-0.1.develop.20220419072414git2577699.el8.noarch                                                                                                                                         1/2 
  Running scriptlet: foreman-selinux-3.3.0-0.1.develop.20220419072414git2577699.el8.noarch                                                                                                                                         1/2 
libsepol.context_from_record: type foreman_container_port_t is not defined (No such file or directory).
libsepol.context_from_record: could not create context structure (Invalid argument).
libsepol.port_from_record: could not create port structure for range 2375:2375 (tcp) (Invalid argument).
libsepol.sepol_port_modify: could not load port range 2375 - 2375 (tcp) (Invalid argument).
libsemanage.dbase_policydb_modify: could not modify record value (Invalid argument).
libsemanage.semanage_base_merge_components: could not merge local modifications into policy (Invalid argument).
OSError: [Errno 22] Invalid argument

  Running scriptlet: foreman-selinux-3.2.0-1.el8.noarch                                                                                                                                                                            2/2 
/sbin/restorecon: SELinux: Could not get canonical path for /etc/puppet/node.rb restorecon: No such file or directory.

  Cleanup          : foreman-selinux-3.2.0-1.el8.noarch                                                                                                                                                                            2/2 
  Running scriptlet: foreman-selinux-3.3.0-0.1.develop.20220419072414git2577699.el8.noarch                                                                                                                                         2/2 
/sbin/restorecon: SELinux: Could not get canonical path for /etc/puppet/node.rb restorecon: No such file or directory.

  Running scriptlet: foreman-selinux-3.2.0-1.el8.noarch                                                                                                                                                                            2/2 
  Verifying        : foreman-selinux-3.3.0-0.1.develop.20220419072414git2577699.el8.noarch                                                                                                                                         1/2 
  Verifying        : foreman-selinux-3.2.0-1.el8.noarch                                                                                                                                                                            2/2 

Upgraded:
  foreman-selinux-3.3.0-0.1.develop.20220419072414git2577699.el8.noarch                                                                                                                                                                

Complete!

[root@centos8-stream-foreman-3-2 ~]# semanage port -l |grep container
foreman_container_port_t       tcp      2376, 2375

[root@centos8-stream-foreman-3-2 ~]# foreman-selinux-enable 
libsepol.context_from_record: type foreman_container_port_t is not defined (No such file or directory).
libsepol.context_from_record: could not create context structure (Invalid argument).
libsepol.port_from_record: could not create port structure for range 2375:2375 (tcp) (Invalid argument).
libsepol.sepol_port_modify: could not load port range 2375 - 2375 (tcp) (Invalid argument).
libsemanage.dbase_policydb_modify: could not modify record value (Invalid argument).
libsemanage.semanage_base_merge_components: could not merge local modifications into policy (Invalid argument).
OSError: [Errno 22] Invalid argument

[root@centos8-stream-foreman-3-2 ~]# vi /usr/sbin/foreman-selinux-enable # apply patch ;)

[root@centos8-stream-foreman-3-2 ~]# foreman-selinux-enable 

[root@centos8-stream-foreman-3-2 ~]# semanage port -l |grep container
[root@centos8-stream-foreman-3-2 ~]# 
```

So IMHO does what it should ;-)

@lzap @ehelms thoughts?

**Comment by evgeni** [comment]
This should either `Refs #34730` or get an own redmine tho.

**Comment by evgeni** [comment]
Originally reported here: https://community.theforeman.org/t/luna-nightly-rpm-pipeline-354-failed/28201/3

**Review by lzap — APPROVED** [review]
Oh yeah, this is what needs to be done. I usually left a comment TODO/FIXME to remove the line in the future version. But the first upgrade must be done before policy is loaded.

**Comment by ehelms** [comment]
Thanks @upadhyeammit

**Pull request merged**